### DISCUSSION TITLE
Can I parse RSS with this library?

### DISCUSSION THREAD

**Discussion by erayerdin** [renggli/dart-xml#176] [Q&A] [answered]
Does this library expect and validate some XML specific features such as `<?xml version="1.0"?>` header? Does it throw exception if it is not valid?

RSS (correct me if I'm wrong) are valid XMLs. Some RSS feeds, however, oddly don't provide `<?xml version="1.0"?>` header. [Here is an example which doesn't contain this header](https://evrimagaci.org/rss.xml).

Would it continue parsing or would it throw an error?

**Comment by renggli** [comment, marked as answer]
Yes, that example works just fine:

```dart
import 'dart:convert';
import 'dart:io';

import 'package:xml/xml.dart';

Future<void> main(List<String> arguments) async {
  final url = Uri.parse('https://evrimagaci.org/rss.xml');
  final request = await HttpClient().getUrl(url);
  final response = await request.close();
  final input = await response.transform(utf8.decoder).join();
  final document = XmlDocument.parse(input);
  print(document.toXmlString(pretty: true));
}
```

The [XML declaration is optional](https://www.w3.org/TR/xml/#NT-prolog) according to the standard, so this library doesn't require it. `XmlDocument.parse` is very strict with XML that does not follow the standard and throws an `XmlParserException`, i.e. when the tags are not balanced, when there are invalid characters, missing closed attributes, ... With the streaming parser you get more control and can resume at any point, but it is also more complicated to use.

**Reply by erayerdin** [reply]
Thanks for quick clarification @renggli.
